reconcile/statuspage/page.py

```diff
@@ -64,47 +45,18 @@ def desired_component_status(self) -> Optional[str]:
     class Config:
         arbitrary_types_allowed = True
 
-
-class StatusPageProvider(ABC):
-    """
-    Provider specific status page reconcile implementation.
-    """
-
-    @abstractmethod
-    def get_current_page(self) -> "StatusPage": ...
-
-    @abstractmethod
-    def apply_component(self, dry_run: bool, desired: StatusComponent) -> None: ...
-
-    @abstractmethod
-    def delete_component(self, dry_run: bool, component_name: str) -> None: ...
-
-
-def build_status_page(
-    page: StatusPageV1,
-) -> "StatusPage":
-    """
-    Translate a desired state status page into a status page object.
-    """
-    return StatusPage(
-        name=page.name,
-        components=[
-            build_status_page_component(component=c) for c in page.components or []
-        ],
-    )
-
-
-def init_provider_for_page(
-    page: StatusPageV1,
-    token: str,
-    component_binding_state: ComponentBindingState,
-) -> StatusPageProvider:
-    """
-    Initialize a status page provider for a given status page.
-    """
-    if page.provider in _PROVIDERS:
-        return _PROVIDERS[page.provider](page, token, component_binding_state)
-    raise ValueError(f"provider {page.provider} is not supported")
+    @staticmethod
+    def init_from_page_component(component: StatusPageComponentV1) -> "StatusComponent":
+        status_configs = [
+            build_status_provider_config(cfg) for cfg in component.status_config or []
+        ]
+        return StatusComponent(
```

Comment:
same here can use classmethod and `Self`

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4343/commits/7549eabf004bae065958b41195a124820ee1ea8e